src/repr/benches/packed.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::NaiveTime;
+use criterion::{criterion_group, criterion_main, Criterion, Throughput};
+use mz_repr::adt::datetime::PackedNaiveTime;
+use mz_repr::adt::interval::{Interval, PackedInterval};
+
+fn bench_interval(c: &mut Criterion) {
+    let mut group = c.benchmark_group("PackedInterval");
+    group.throughput(Throughput::Elements(1));
+
+    const INTERVAL: Interval = Interval::new(1, 1, 0);
+    group.bench_function("encode", |b| {
+        b.iter(|| {
+            let packed = PackedInterval::from(INTERVAL);
```

Comment:
I might be totally off, but don't you need to `black_box` `INTERVAL`, too? Otherwise the optimizer might just constant-fold the expresion?

Reply:
That totally makes sense, the benchmarks also seemed too fast and the numbers look more reasonable now, thanks!